src/SprykerEco/Zed/Optivo/Business/OptivoBusinessFactory.php

```diff
@@ -1,86 +1,211 @@
+
+    /**
+     * @return \SprykerEco\Zed\Optivo\Dependency\Facade\OptivoToMoneyFacadeInterface
+     */
+    protected function getMoneyFacade(): OptivoToMoneyFacadeInterface
+    {
+        return $this->getProvidedDependency(OptivoDependencyProvider::FACADE_MONEY);
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Optivo\Dependency\Facade\OptivoToLocaleFacadeInterface
+     */
+    protected function getLocaleFacade(): OptivoToLocaleFacadeInterface
+    {
+        return $this->getProvidedDependency(OptivoDependencyProvider::FACADE_LOCALE);
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Optivo\Dependency\Facade\OptivoToSalesFacadeInterface
+     */
+    protected function getSalesFacade(): OptivoToSalesFacadeInterface
+    {
+        return $this->getProvidedDependency(OptivoDependencyProvider::FACADE_SALES);
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Optivo\Business\Api\Request\RequestUrlBuilderInterface
+     */
+    protected function createRequestUrlBuilder(): RequestUrlBuilderInterface
+    {
+        return new RequestUrlBuilder($this->getConfig());
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Optivo\Business\Api\Response\ResponseConverterInterface
+     */
+    protected function createResponseConverter(): ResponseConverterInterface
+    {
+        return new ResponseConverter();
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Optivo\Business\Api\Adapter\Http\HttpAdapterInterface
+     */
+    protected function createGuzzleAdapter(): HttpAdapterInterface
+    {
+        return new Guzzle($this->getConfig());
+    }
+
     /**
      * @return \SprykerEco\Zed\Optivo\Business\Api\Adapter\OptivoApiAdapterInterface
      */
-    protected function createOptivoApiAdapter()
+    protected function createOptivoApiAdapter(): OptivoApiAdapterInterface
     {
         return new OptivoApiAdapter(
-            $this->createGuzzle(),
+            $this->createGuzzleAdapter(),
             $this->createRequestUrlBuilder(),
             $this->createResponseConverter()
         );
     }
 
     /**
-     * @return \SprykerEco\Zed\Optivo\Business\Api\Adapter\Http\HttpAdapterInterface
+     * @return \SprykerEco\Zed\Optivo\Business\Mapper\Customer\CustomerMapperInterface
      */
-    protected function createGuzzle()
+    protected function createCustomerMapper(): CustomerMapperInterface
     {
-        return new Guzzle($this->getConfig());
+        return new CustomerMapper($this->getConfig(), $this->getLocaleFacade());
     }
 
     /**
-     * @return \SprykerEco\Zed\Optivo\Business\Api\Request\RequestUrlBuilderInterface
+     * @return \SprykerEco\Zed\Optivo\Business\Mapper\Customer\CustomerMapperInterface
      */
-    protected function createRequestUrlBuilder()
+    protected function createCustomerNewsletterMapper(): CustomerMapperInterface
     {
-        return new RequestUrlBuilder(
-            $this->getConfig()
-        );
+        return new CustomerNewsletterMapper($this->getConfig(), $this->getLocaleFacade());
```

Comment:
I think it will look better if you indent this with new lines.

Reply:
I think it is less then 120 chars, so it is ok